packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -166,10 +168,40 @@ export function trapEventForPluginEventSystem(
   );
 
   const rawEventName = getRawEventName(topLevelType);
+  let fbListener;
+
+  // When legacyFBSupport is enabled, it's for when we
+  // want to add a one time event listener to a container.
+  // This should only be used with enableLegacyFBPrimerSupport
+  // due to requirement to provide compatibility with
+  // internal FB www event tooling. This works by removing
+  // the event listener as soon as it is invoked. We could
+  // also attempt to use the {once: true} param on
+  // addEventListener, but that requires support and some
+  // browsers do not support this today, and given this is
+  // to support legacy code patterns, it's likely they'll
+  // need support for such browsers.
+  if (enableLegacyFBPrimerSupport && legacyFBSupport) {
+    const originalListener = listener;
```

Comment:
What happens if Primer code ever calls `stopImmediatePropagation`? This handler just hangs out until the next click event? (Not that this seems likely to happen I guess)

Reply:
It doesn't do that, thankfully.